compiler/ballerina-lang/src/main/resources/compiler.properties

```
@@ -1109,6 +1109,9 @@ error.invalid.error.constructor.rest.detail.arg.on.detail.type.with.individual.f
   error constructor does not accept additional detail args ''{0}'' when error detail type ''{1}'' contains individual \
   field descriptors
 
+error.unknown.error.detail.field.in.binding.pattern=\
+  unknown error detail field ''{0}'' in binding pattern
```

How about something like
```suggestion
  invalid field binding pattern; cannot bind undefined error detail field ''{0}''
```

or

```suggestion
  invalid field binding pattern; cannot bind an undefined error detail field
```

since we log at the specific location.

---

How about 

```
invalid error field binding pattern; cannot bind undefined error detail field ''{0}''
```

https://ballerina.io/spec/lang/master/#error-field-binding-patterns

---

Went ahead with `cannot bind undefined error detail field ''{0}''` as per our offline discussion.
Created issue #34641 to align other places.